expression/builtin_arithmetic.go

```diff
@@ -729,27 +729,41 @@ func (s *builtinArithmeticIntDivideIntSig) evalInt(row chunk.Row) (int64, bool,
 	return ret, err != nil, errors.Trace(err)
 }
 
-func (s *builtinArithmeticIntDivideDecimalSig) evalInt(row chunk.Row) (int64, bool, error) {
-	a, isNull, err := s.args[0].EvalDecimal(s.ctx, row)
-	if isNull || err != nil {
-		return 0, isNull, errors.Trace(err)
-	}
-
-	b, isNull, err := s.args[1].EvalDecimal(s.ctx, row)
-	if isNull || err != nil {
-		return 0, isNull, errors.Trace(err)
+func (s *builtinArithmeticIntDivideDecimalSig) evalInt(row chunk.Row) (ret int64, isNull bool, err error) {
+	sc := s.ctx.GetSessionVars().StmtCtx
+	var num [2]*types.MyDecimal
+	for i, arg := range s.args[0:2] {
```

Comment:
Should we return an error if len(args) is not 2?

Reply:
argument validation should be done in this phase.

Reply:
So use `s.args` is also safe here?

Reply:
oh， I forgot it. use s.args is enough.